app/models/service_terraform_template.rb

```diff
@@ -76,7 +77,25 @@ def check_refreshed(_action)
   private
 
   def get_job_options(action)
-    options[job_option_key(action)].deep_dup
+    case action.downcase
+    when 'retirement'
+      retire_job_options = options[job_option_key('retirement')].deep_dup
+      retire_job_options[:action] = action
+      prov_job_options = options[job_option_key('provision')].deep_dup
+      # if not already available in retirement options, (won't be available for terraform)
+      # copy extra_vars,execution_ttl,verbosity,credentials,etc from provision action
+      job_options = prov_job_options.deep_merge(retire_job_options)
+    else
```

Comment:
This doesn't seem right, shouldn't we set what we need in the options for the retirement action so we don't have to do this?

Reply:
With new changes, we will only use `vars` from Provision Job. We still need `vars` values from Provision job, because the Retirement option only has default value in ServiceTemplates the default values, we want the user-modified values from the Provision Job.